astropy/utils/data.py

```diff
@@ -1089,6 +1105,8 @@ def download_file(remote_url, cache=False, show_progress=True, timeout=None,
         timeout = conf.remote_timeout
     if sources is None:
         sources = [remote_url]
+    if http_headers is None:
+        http_headers = {'User-Agent': 'astropy','Accept': '*/*'}
```

Comment:
I still don't like this but I'll defer to @aarchiba for a second opinion.

Reply:
So just to be clear, @pllim, you're proposing instead having this in the iers.py `download_file` instead of this one?

I'm open to that but I think this is actually better because it will make it possible to keep better track of Astropy on remote services for the long-term (and my understanding is that you're "supposed" to do this according to some HTTP recommendations, but could be wrong about that).  So I agree @aarchiba is the deciding vote here.

Reply:
Sorry to be slow getting back to you on this, I wanted to read more about what good citizenship means in terms of HTTP headers. What is the correct User-Agent for a (semi-)unattended script? Is the Accepts claiming something peculiar? But the fact that `requests` apparently sends these headers by default on every request leads me to think that that is probably a good idea.

Reply:
So, a third-party package that is using `astropy.utils.data` to download files would be telling the service provider that the request come from `astropy`. Certainly, they can pass in their own header but I have a feeling that 95% of the people would just use the default. Is this side effect desirable?

Reply:
I think I'm fine with the current behavior as long as it is made very clear in the docstrings (right now they do not clearly expose what the default behavior is). 

As another idea, what if we made a config item for "package_user_agent" or something, and set it to "astropy" for astropy core? Then the default can be "None" or no user-agent name, but we give other people that use this functionality within other packages the ability to easily control their user-agent name?

Reply:
I'm coming around to the config item idea being the better solution, however, in the interest of time and given that things around this are broken in master, I would be fine with making an issue for that and merging this as is (after adding the default behavior to the docstring).